### DISCUSSION TITLE
is there a way to distinguish the regular font and pseudo bold in pymupdf

### DISCUSSION THREAD

**Discussion by zhangxiaojiawow** [pymupdf/PyMuPDF#2881] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
There are some text which is pseudo bold font in my pdf file, I want to distinguish it with the regular font text. currently, using the font size，flag and  font name, I can't achieve this goal.

**Comment by JorjMcKie** [comment, marked as answer]
This is a Discussions item and no issue.

There are multiple methods to let non-bold text **_appear_** bold:
1. Write the same text twice, with a small offset the second time. The PDF author may have done this character-by-character or in greater chunks (word-by-word, line-by-line, etc.). This can be detected as doubled text in extractions, ("aaabbbccc", or "abcabcabc" etc.) and (largely) overlapping text positions.
2. Thicken the single character borders: normally characters are written by only filling its interior with a fill color. In addition, one can add a stroke color to also write the characters' border lines. This cannot be detected in normal text extraction (and this will not change in the foreseeable future). But you can use `page.get_texttrace()`. This is a low level text extraction, which returns separate text spans for the fill color version of text and the stroke color version. Each spans contains a "type" key in the dictionary, which is normally 0 (fill color) or "1" (stroke color). Please consult its documentation.

Here is an image explaining point 2:
![image](https://github.com/pymupdf/PyMuPDF/assets/8290722/7d491f1c-3ceb-4dac-b5b0-04e6b5f43e73)

Imagine both colors are black, then render mode 2 would appear like bold.

**Reply by zhangxiaojiawow** [reply]
Hello, @JorjMcKie :
Thanks for your quick reply,  this should be a discussion. Your method is valid in my case(actually the second situations).
